### PR TITLE
Make TPL code Silverlight 5 compatible

### PR DESCRIPTION
**Pull request by iskiselev**
Per discussion in #66, Silverlight 4 support was dropped and added TPL Task support to SL5 version. #70 just disabled TPL Task support for Silverlifht version, this PR re-enable it.

### CODE REVIEW

**Comment by kzu** [comment]
Do we still need this? The SL is already 5.0, not sure we need to rebase this PR to get it in?

**Comment by iskiselev** [comment]
I think we still need this. I've rebased PR.

File: UnitTests.Silverlight/Moq.Silverlight.Tests.csproj [hunk 1]
```diff
@@ -239,4 +242,11 @@
       </FlavorProperties>
     </VisualStudio>
   </ProjectExtensions>
+  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
```
Comment (kzu) [inline UnitTests.Silverlight/Moq.Silverlight.Tests.csproj]: I'm fine moving to Castle nuget, but let us not go down the [old nuget restore rabbit whole](http://www.cazzulino.com/ultimate-cross-platform-nuget-restore.html) ;)

**Comment by iskiselev** [comment]
Removed Nuget restore.

File: Source.Silverlight/Moq.Silverlight.csproj [hunk 2]
```diff
@@ -21,6 +21,8 @@
     <AssemblyOriginatorKeyFile>..\Moq.snk</AssemblyOriginatorKeyFile>
     <SignManifests>false</SignManifests>
     <OldToolsVersion>4.0</OldToolsVersion>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
```
Comment (kzu) [inline Source.Silverlight/Moq.Silverlight.csproj]: these two properties are part of nuget restore and should be gone too

File: UnitTests.Silverlight/Moq.Silverlight.Tests.csproj [hunk 3]
```diff
@@ -27,6 +27,8 @@
     <AssemblyOriginatorKeyFile>..\Moq.snk</AssemblyOriginatorKeyFile>
     <SignManifests>false</SignManifests>
     <OldToolsVersion>4.0</OldToolsVersion>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
```
Comment (kzu) [inline UnitTests.Silverlight/Moq.Silverlight.Tests.csproj]: ditto

**Comment by kzu** [comment]
Hm.. the main csproj has them, so merging :)

**Comment by iskiselev** [comment]
I've created fix without them. Should I create new PR or will you leave this NuGet properties?

**Comment by kzu** [comment]
leave them there, no worries.

thanks for the fix!

**Pull request merged**